consul/src/main/scala/io/buoyant/consul/v1.scala

```diff
@@ -44,7 +49,7 @@ package object v1 {
     Try(mapper.readValue[T](bytes, begin, end - begin))
   }
 
-  object Api {
+  object CatalogApi {
```

Comment:
You mean classes, right? Or objects as well? I may miss something but with regards to objects I see only 1 'generalization option' - add a type parameter to the base one and implement `apply` there...